activesupport/lib/active_support/testing/composite_filter.rb

```diff
@@ -17,11 +17,8 @@ def derive_regexp(filter)
           filter =~ %r%/(.*)/% ? Regexp.new($1) : filter
         end
 
-        def derive_line_filters(patterns)
-          patterns.map do |file_and_line|
-            file, line = file_and_line.split(':')
-            Filter.new(@runnable, file, line) if file
-          end
+        def build_filter(file, line)
+          Filter.new(@runnable, file, line) if file
         end
```

Comment:
Think this would do:

```ruby
def derive_line_filters(patterns)
  patterns.flat_map do |file_and_line|
    file, *lines = file_and_line.split(':')
    lines.map { |line| Filter.new(@runnable, file, line) } if file
  end
end
```